### DISCUSSION TITLE
Trigger macro on time stamp interval click

### DISCUSSION THREAD

**Discussion by tak5haka** [vigoren/foundryvtt-simple-calendar#412] [Q&A]
Dumb question, but is there a way to trigger a custom macro when the "dawn", "midday", "dusk" or "midnight" buttons are clicked? I currently have a macro that advances the time preset, but would like to save time (and space on my hotbar) by having this trigger when the preset is clicked.

Cheers

**Comment by vigoren** [comment]
Hey!

As of right now there is no way to trigger a macro when those buttons are clicked. Simple calendar does emit a hook with the day changes, though nothing in the data returned by that hook indicates if one of the preset time off day buttons was clicked.

**Reply by tak5haka** [reply]
Ah, okay! Follow-up question, is there a way to grab the hour value from the DateTimeChange data? I've tried different ways (eg data.date.hour), but doesn't seem to always work for some reason. Probably because I don't really know what I am doing :)

**Reply by vigoren** [reply]
There is yes.

```
Hooks.on(SimpleCalendar.Hooks.DateTimeChange, (data) => {
   console.log(data.date.hour);
 });
```

This code needs to be run before you change the date in Simple Calendar, then when you change the date the browser console will show a line with the current hour.


How are you trying to execute the code?

**Comment by tak5haka** [comment]
Brilliant, thanks. I have a macro that is listening for a change to the
date or time and then runs some calculations to generate a chat card with
information about that day. Seems to be working now, so thanks for your
help.
